tx-details-storage/src/drivers/postgres.rs

```diff
@@ -0,0 +1,268 @@
+    async fn create_tables(pool: &PgPool) -> Result<()> {
+        sqlx::query(
+            "
+            CREATE TABLE IF NOT EXISTS transactions (
+                transaction_hash VARCHAR PRIMARY KEY,
+                sender_account_id VARCHAR NOT NULL,
+                block_height NUMERIC(20,0) NOT NULL,
+                transaction_details BYTEA NOT NULL
+            );
+            ",
+        )
+        .execute(pool)
+        .await?;
+
+        sqlx::query(
+            "
+            CREATE TABLE IF NOT EXISTS receipts (
+                receipt_id VARCHAR PRIMARY KEY,
+                parent_transaction_hash VARCHAR NOT NULL,
+                receiver_id VARCHAR NOT NULL,
+                block_height NUMERIC(20,0) NOT NULL,
+                block_hash VARCHAR NOT NULL,
+                shard_id BIGINT NOT NULL
+            );
+            ",
+        )
+        .execute(pool)
+        .await?;
+
+        sqlx::query(
+            "
+            CREATE TABLE IF NOT EXISTS outcomes (
+                outcome_id VARCHAR PRIMARY KEY,
+                parent_transaction_hash VARCHAR NOT NULL,
+                receiver_id VARCHAR NOT NULL,
+                block_height NUMERIC(20,0) NOT NULL,
+                block_hash VARCHAR NOT NULL,
+                shard_id BIGINT NOT NULL
+            );
+            ",
+        )
+        .execute(pool)
+        .await?;
+
+        sqlx::query(
+            "
+            CREATE TABLE IF NOT EXISTS meta (
+                indexer_id VARCHAR PRIMARY KEY,
+                last_processed_block_height NUMERIC(20,0) NOT NULL
+            );
+            ",
+        )
+        .execute(pool)
+        .await?;
+
+        Ok(())
+    }
+}
+
+#[async_trait]
+impl Storage for PostgresTxDetailsStorage {
+    async fn save_tx(&self, key: &str, data: Vec<u8>, block_height: u64) -> Result<()> {
+        sqlx::query(
+            "
+            INSERT INTO transactions (transaction_hash, sender_account_id, block_height, transaction_details)
+            VALUES ($1, $2, $3, $4)
+            ON CONFLICT (transaction_hash) DO NOTHING;
+            ",
+        )
+        .bind(key)
+        .bind("default_sender") // TODO: use it to determine the shard id to store the data
+        .bind(BigDecimal::from(block_height))
+        .bind(data)
+        .execute(&self.pool)
+        .await?;
+        Ok(())
+    }
+
+    async fn retrieve_tx(&self, key: &str) -> Result<Vec<u8>> {
+        let row = sqlx::query(
+            "
+            SELECT transaction_details
+            FROM transactions
+            WHERE transaction_hash = $1;
+            ",
+        )
+        .bind(key)
+        .fetch_one(&self.pool)
+        .await?;
+
+        let data: Vec<u8> = row.get("transaction_details");
+        Ok(data)
+    }
+
+    async fn save_receipts(
+        &self,
+        receipts: Vec<readnode_primitives::ReceiptRecord>,
+        block_height: u64,
+    ) -> Result<()> {
+        for receipt in receipts {
```

Comment:
I think we should use bulk_insert. Saving one by one will increase the load on the database and will work very slowly.